### PR TITLE
Monadfix

### PR DESCRIPTION
**Pull request by dvlasits**

### CODE REVIEW

File: tests/test.ml [hunk 1]
```diff
@@ -161,3 +161,8 @@ let () =
   assert (append e e = e);
   assert (append {First {Any_Int}} { first = Some 2 } e = { first = Some 2 });
   assert (append {First {Any_Int}} { first = Some 2 } { first = Some 3 } = { first = Some 2 })
+
+  let () =
+  let open Imp.Control in
+  let f = fun x -> Some 5 in
+    assert ((mfix f) = (Some 5))
```
Comment (pxeger) [inline tests/test.ml]: Add the missing newline please, and configure your editor to add the automatically

**Review by pxeger — CHANGES_REQUESTED** [review]
Does MonadFix even particularly make sense in a strict language? Do you have an envisioned use-case? I think we should avoid adding too many immensely-abstract concepts if they're not really useful in practice.

**Review by pxeger — APPROVED** [review]

**Comment by dvlasits** [comment]
> Does MonadFix even particularly make sense in a strict language? Do you have an envisioned use-case? I think we should avoid adding too many immensely-abstract concepts if they're not really useful in practice.

I guess whenever you use lazy features

**Comment by dvlasits** [comment]
> > Does MonadFix even particularly make sense in a strict language? Do you have an envisioned use-case? I think we should avoid adding too many immensely-abstract concepts if they're not really useful in practice.
> 
> I guess whenever you use lazy features

streams are probably a nice use-case

**Comment by pxeger** [comment]
Yeah but it does actually work with streams?

**Comment by dvlasits** [comment]
> Yeah but it does actually work with streams?

my plan to give that a go now, but surely ...

**Pull request merged**